src/hmem.c

```diff
@@ -285,12 +286,29 @@ static ssize_t ofi_copy_mr_iov(struct ofi_mr **mr, const struct iovec *iov,
 
 		if (mr && mr[i]) {
 			hmem_iface = mr[i]->iface;
+			hmem_flags = mr[i]->flags;
 			hmem_device = mr[i]->device;
+			hmem_data = mr[i]->hmem_data;
 		} else {
 			hmem_iface = FI_HMEM_SYSTEM;
+			hmem_flags = 0;
 			hmem_device = 0;
+			hmem_data = NULL;
 		}
-		if (dir == OFI_COPY_BUF_TO_IOV)
+
+		if (hmem_iface == FI_HMEM_CUDA && (hmem_flags & OFI_HMEM_DATA_GDRCOPY_HANDLE)) {
+			/**
+			 * TODO: Fine tune the max data size to switch from gdrcopy to cudaMemcpy
+			 * Note: buf must be on the host since gdrcopy does not support D2D copy
+			 */
+			if (dir == OFI_COPY_BUF_TO_IOV)
```

Comment:
hmmm. so there is no way to make cuda_gdrcopy* called by ofi_hmem_copy* ?

Reply:
Got it , makes sense. I see it is wrapped inside ofi_mr_iov_copy, which should be fine